k2/python/csrc/torch/fsa.cu

```diff
@@ -70,9 +61,42 @@ static void PybindFsaUtil(py::module &m) {
       "returned FSA is an acceptor");
 }
 
+static void PybindDenseFsaVec(py::module &m) {
+  using PyClass = DenseFsaVec;
+  py::class_<PyClass> pyclass(m, "DenseFsaVec");
+  // We do not need to access its members in Python
+
+  // TODO(fangjun): add docstring for this funciton
+  pyclass.def(
+      py::init([](torch::Tensor scores, torch::Tensor sizes) -> DenseFsaVec * {
+        // remove the contiguous check once the following comment
```

Comment:
OK.  Bear in mind that on the Python side we'll need some convenience wrappers to get things into this format.  Look at the comments in the C++ to understand stuff about the -infinity's.  Basically the original format will be: a tensor of nnet outputs (say: shape=(batch_size,T,dim)); and 3 pieces of information for each supervised segment: batch_index, start_time, duration.

Reply:
I was thinking that it might be better to provide a single interface to construct `DenseFsaVec` and the supervision `FsaVec` together, which is less error-prone.

The `SupervisionSegment` from `lhotse` is defined as
https://github.com/pzelasko/lhotse/blob/master/lhotse/supervision.py#L8
```python
class SupervisionSegment:
    start: Seconds
    duration: Seconds
    text: Optional[str] = None
```

I guess `k2` is not going to depend on `lhotse`. I'd propose to add a supervision class for k2.

Reply:
Mm.  The supervision will really just be an FsaVec.  (Its 1st index, the fsa_idx, corresponds to the 1st index in the DenseFsaVec).

Reply:
... The creation of the supervision FSAs, and their I/O, will rely on fairly general FSA functions.  The only special thing is that in many cases we won't need to retain autograd info for the graph creation.